### PR TITLE
remove execCreate & execStart from job

### PR DESCRIPTION
**Pull request by LK4D4**
Also removed the function ExecConfigFromJob

Closes #12472

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM, as long as janky agrees

File: api/server/server.go [hunk 1]
```diff
@@ -1393,35 +1391,27 @@ func (s *Server) postContainerExecCreate(eng *engine.Engine, version version.Ver
...
-	// Parse warnings from stderr
-	scanner := bufio.NewScanner(warnings)
-	for scanner.Scan() {
-		outWarnings = append(outWarnings, scanner.Text())
-	}
 
 	return writeJSON(w, http.StatusCreated, &types.ContainerExecCreateResponse{
-		ID:       engine.Tail(stdoutBuffer, 1),
-		Warnings: outWarnings,
+		ID: id,
```
Comment (dqminh) [inline api/server/server.go]: This removes stderr content -> `Warnings`, dont we want that as well ?

File: api/server/server.go [hunk 2]
```diff
@@ -1455,21 +1448,20 @@ func (s *Server) postContainerExecStart(eng *engine.Engine, version version.Vers
...
+		stdout = outStream
+		stderr = errStream
 	}
 	// Now run the user process in container.
-	job.SetCloseIO(false)
-	if err := job.Run(); err != nil {
-		fmt.Fprintf(errOut, "Error starting exec command in container %s: %s\n", name, err)
+
+	if err := s.daemon.ContainerExecStart(execName, stdin, stdout, stderr); err != nil {
+		fmt.Fprintf(stderr, "Error starting exec command in container %s: %s\n", execName, err)
```
Comment (duglin) [inline api/server/server.go]: If we we set execStartCheck.Detach to true, will stderr be nil here?

Comment (duglin) [inline api/server/server.go]: I think you need to reintroduce errOut and have it be either the http stream or os.Stderr to match the old way it was done. Of course, I don't like writing to os.Stderr, we should just log it instead.

Comment (LK4D4) [inline api/server/server.go]: I think we can ignore old-way on daemon side. We can just log error every time.

File: api/server/server.go (hunk 1 above)
Comment (LK4D4) [inline api/server/server.go]: Yeah, I think we can ignore `stderr` now for such purposes.

File: api/server/server.go (hunk 2 above)
Comment (duglin) [inline api/server/server.go]: Yup - log it but in the case of being attached we need to send it back on the wire, right?

**Comment by duglin** [comment]
LGTM once janky is happy

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**